llvm/lib/CodeGen/PHIEliminationUtils.cpp

```diff
@@ -13,6 +13,19 @@
 
 using namespace llvm;
 
+// Returns true if MBB contains an INLINEASM_BR instruction that may
+// branch to SuccMBB, requiring specialized copy placement.
+static bool hasInlineAsmBrToSuccessor(MachineBasicBlock *MBB,
+                                      MachineBasicBlock *SuccMBB) {
+  if (!SuccMBB->isInlineAsmBrIndirectTarget())
+    return false;
+
+  for (const MachineInstr &MI : *MBB)
```

Comment:
Does this need to scan the whole block?

Reply:
Not sure if there's a best alternative, INLINEASM_BR doesn't seem to be a terminator and we might introduce copies between INLINEASM_BR and the jump to its fallthrough during codegen.

Reply:
Should we reverse-walk the BB at least?
I'd expect the inlineasm br to be towards the end of the basic block.

Reply:
> Should we reverse-walk the BB at least?

Moved to reverse walk (I wouldn't expect the indirect target BB to be the same one where the INLINEASM_BR lies on average, though better than nothing).

> It is required to be in the terminators

Tested whether INLINEASM_BR appears in `terminators()`, and it doesn't seem to be the case. I agree it should be (like its IR counterpart), I'll take a look as a follow-up.

Reply:
> Does this need to scan the whole block?

Yes since INLINEASM_BR is not a Terminator.